.ci/linters/rd/condition_class_docs_check.R

```diff
@@ -0,0 +1,51 @@
+# Ensure that data.table condition classes in code match documentation
+condition_classes_documentation_linter = function(rd_file) {
+  if (!grepl("\\name{datatable-condition-classes}", readChar(rd_file, 100L), fixed = TRUE)) return(invisible())
+
+  # Find condition classes in R code 
+  walk_r_ast_for_classes = function(expr) {
+    if (is.call(expr) && length(expr) >= 2L && identical(expr[[1L]], quote(stopf)) && is.character(class_arg <- expr[["class"]]) && startsWith(class_arg, "dt_")) {
```

Comment:
I think that's redundant, too -- the only exception is an empty `stopf()` call, which is unlikely in practice.

```suggestion
    if (is.call(expr) && identical(expr[[1L]], quote(stopf)) && is.character(class_arg <- expr[["class"]]) && startsWith(class_arg, "dt_")) {
```